src/State/README.md

```diff
@@ -0,0 +1,667 @@
+
+// updateKey :: String -> a -> State Object ()
+const updateKey = key => val =>
+  modify(assoc(key, val))
+
+// addToSub :: Number -> String Object Number
+const addToSub = x =>
+  getKey('sub')
+    .map(add(x))
+
+const calcTax =
+  getKey('tax')
+    .map(multiply)
+    .ap(getKey('sub'))
+
+// applyTax :: State Object ()
+const applyTax =
+  calcTax
+    .chain(addToSub)
+    .map(round)
+    .chain(updateKey('total'))
+
+applyTax
+  .execWith(data)
+//=> { tax: 0.084, sub: 34.07, total: 37.91 }
+```
+
+### chain
+```haskell
+State s a ~> (a -> State s b) -> State s b
+```
+
+As a means to combine stateful computations, `chain` is used to sequence
+state transactions that either read from or write to the state. `chain` takes
+a unary function that must return a new `State` instance. `chain` returns a new
+`State` instance that will apply the computation when run.
+
+```js
+const { get, modify } = require('crocks/State')
+
+// add :: Number -> State Number ()
+const add = x =>
+  modify(y => x + y)
+
+// multiply :: Number -> State Number ()
+const multiply = x =>
+  modify(y => x * y)
+
+// double :: () -> State Number ()
+const double = () =>
+  get()
+    .chain(add)
+
+// square :: () -> State Number ()
+const square = () =>
+  get()
+    .chain(multiply)
+
+add(10)
+  .execWith(10)
+//=> 20
+
+add(10)
+  .chain(double)
+  .execWith(10)
+//=> 40
+
+add(10)
+  .chain(square)
+  .execWith(10)
+//=> 400
+
+add(10)
+  .chain(double)
+  .chain(square)
+  .execWith(10)
+//=> 1600
+```
+
+### runWith
+```haskell
+State s a ~> s -> Pair a s
+```
+
+`State` is a lazy datatype that is requires a value for it's state portion to
+be run. A given `State` instance provides a `runWith` method that accepts a
+value to run the instance with. The value must be a member of the type that the
+given `State` instance is fixed to in it's state portion, `s`.
+
+`runWith`, when called with a value, will run the state transition with the given
+value as the initial state and will return the resulting `Pair` with the
+resultant in the `fst` (first) and the state in the `snd` (second).
+
+
+```js
+const State = require('crocks/State')
+const { get, put } = State
+
+const K = require('crocks/combinators/constant')
+
+// swap :: s -> s -> State s s
+const swap = x => old =>
+  put(x)
+    .chain(K(State.of(old)))
+
+//update :: s -> State s s
+const update = x =>
+  get()
+    .chain(swap(x))
+
+update(45)
+  .runWith(100)
+//=> Pair(100, 45)
+```
+
+### evalWith
+```haskell
+State s a ~> s -> a
+```
+
+`State` is a lazy datatype that requires a value for it's state portion to
+be run. A given `State` instance provides an `evalWith` method that accepts a
+value to run the instance with. The value must be a member of the type that the
+given `State` instance is fixed to in it's state portion, `s`.
+
+`evalWith`, when called with a value, will run the state transition with the
+given value as the initial state and will return the resulting resultant and
```

Comment:
> return the resulting resultant and will discard the state portion.

return the resulting resultant, discarding the state portion.